.github/workflows/1-Build-Dependency-Image.yaml

```diff
@@ -0,0 +1,82 @@
+name: Build Dependencies Container Image
+run-name: (1) Dependencies Image Build - ${{ github.event.head_commit.message }} - ${{ github.ref_name }}
+
+# This GitHub Actions workflow builds a Docker image for the
+# cfa-epinow2-pipeline-docker project. It consists of two jobs:
+# build_image_dependencies and build_image.
+#
+# - The `build_image_dependencies` job carries out the first part of a
+#   multi-stage build. It downloads and install all the dependencies
+#   listed in the `DESCRIPTION` file. It uses the `Dockerfile-dependencies`
+#   file to build the image.
+#
+#   The built image is then pushed to the corresponding registry.
+#
+#   The process is cached to avoid rebuilding the image if the dependencies
+#   have not changed. This is by hashing the `DESCRIPTION` file and the
+#   `Dockerfile-dependencies` file.
+#
+# - The build_image job builds the final image using the `Dockerfile` file.
+#   It uses the image built in the previous job as a base image.
+#
+#   During the build process, the package is installed and built. Furthermore
+#   the package is checked using `R CMD check` to ensure that it is working
+#   correctly.
+#
+#   Once the image is built, it is pushed to the corresponding registry.
+
+on:
+  push:
+    branches:
+      - main
+      - jk-azure-readiness # temporary
+    paths:
+      - "./Dockerfile-dependencies" # the dockerfile this workflow builds from
+      - "./.github/workflows/build-dependency-image.yaml" # this workflow
+
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "./Dockerfile-dependencies" # the dockerfile this workflow builds from
+      - "./.github/workflows/build-dependency-image.yaml" # this workflow
+
+  workflow_dispatch:
+
+env:
+  # Together, these form: cfaprdbatchcr.azurecr.io/cfa-epinow2-pipeline
+  REGISTRY: cfaprdbatchcr.azurecr.io
+  IMAGE_NAME: cfa-epinow2-pipeline
+
+jobs:
+  Job01-build_image_dependencies:
+    runs-on: cfa-cdcgov # VM based runner serving CFA's cdcgov repos (as opposed to cdcent)
```

Comment:
Good point, I would argue that this workflow should be OK to run using GH hosted runners. Probably more stable.

Reply:
Using the self-hosted runners is a network security constraint placed on us by CDC Network Security. We need them because Github Hosted Runners are not permitted access to our Azure environment.

Reply:
In the mid-term future CFA Tools has a tentative plan to set up a self-hosted kubernetes S/H Runner cluster that can autoscale and recover from faults. No ETA or guarantees, though.